File: src/org/thoughtcrime/securesms/blurhash/BlurHash.java
package org.thoughtcrime.securesms.blurhash;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.Objects;

/**
 * A BlurHash is a compact string representation of a blurred image that we can use to show fast
 * image previews.
 */
public class BlurHash {

  private final String hash;

  private BlurHash(@NonNull String hash) {
    this.hash = hash;
  }

  public static @Nullable BlurHash parseOrNull(@Nullable String hash) {
    if (Base83.isValid(hash)) {
      return new BlurHash(hash);
    }
    return null;
  }

  public @NonNull String getHash() {
    return hash;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    BlurHash blurHash = (BlurHash) o;
    return Objects.equals(hash, blurHash.hash);
  }

  @Override
  public int hashCode() {
    return Objects.hash(hash);
  }
}

File: lib/blurhash/src/main/java/org/signal/blurhash/BlurHash.kt
package org.signal.blurhash

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

/**
 * A BlurHash is a compact string representation of a blurred image that we can use to show fast
 * image previews.
 */
@Parcelize
data class BlurHash(val hash: String) : Parcelable {
  companion object {
    fun parseOrNull(hash: String?): BlurHash? {
      if (Base83.isValid(hash)) {
        return BlurHash(hash!!)
      }
      return null
    }
  }
}
